### ISSUE TITLE
Connect already in progress

### ISSUE THREAD

**Issue by AlainWoeffray** [cloudio-project/cloudio-endpoint-java#16]
For an unknown reason, cloud.io cannot connect to the server and rise these exceptions : 

```
2016-01-05 07:59:57,860 ERROR Exception: Connect already in progress (ch.hevs.cloudio.endpoint.Endpoint$InternalEndpoint)
2016-01-05 08:00:07,868 ERROR Exception: Connect already in progress (ch.hevs.cloudio.endpoint.Endpoint$InternalEndpoint)
2016-01-05 08:00:08,194 ERROR Exception: MqttException (ch.hevs.cloudio.endpoint.Endpoint$InternalEndpoint$1)
```

**Comment by cm0x4D** [comment]
The problem was caused on a device without valid DNS configuration. Still it should not cause an error.

**Comment by cm0x4D** [comment]
Should be fixed by e7b846afe396b67f82583cb4cd6ec8bd70090076
